libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py

```diff
@@ -888,7 +888,10 @@ def write_mcu_config(f):
         build_flags.append('USE_FATFS=no')
         env_vars['DISABLE_SCRIPTING'] = True
     if 'OTG1' in bytype:
-        f.write('#define STM32_USB_USE_OTG1                  TRUE\n')
+        if mcu_type.startswith('STM32H730'):
```

Comment:
I don't think this can be STM32H730 specific. OTG2 is available on other chips and is OTG_HS. We have an hardware that does this with H757 https://github.com/CubePilot/ardupilot/commit/45f8afa6a8bd2d1f69a03ffc1311e0298980429d#diff-00cdd2b90e14e9bac5a19c02a223046a45d1eb38242337320621df067739b84c . For your reference. It also does required changes to use OTG_HS.

Reply:
It seems like OTG2 is being used as OTG1 in H730. I think it might better to do it as a generic define in mcuconf `#define STM32_OTG2_IS_OTG1 1` instead of making it a chip specific thing. This way different chips with similar behaviour can use this as well.

Reply:
ok

Reply:
it requires something in chibios_hwdef since that is where we enable/disable OTG - I have now done this more generally as you suggest